modules/primo/src/Primo/Ting/PrimoStatementRenderer.php

```diff
@@ -0,0 +1,191 @@
+<?php
+
+namespace Primo\Ting;
+
+use Ting\Search\BooleanStatementGroup;
+use Ting\Search\BooleanStatementInterface;
+use Ting\Search\SearchProviderException;
+use Ting\Search\TingSearchCommonFields;
+use Ting\Search\TingSearchFieldFilter;
+use Ting\Search\UnsupportedSearchQueryException;
+
+/**
+ * Renders a list of statements into primo queries.
+ *
+ * @package Primo\Ting
+ */
+class PrimoStatementRenderer {
+  protected $mapping = NULL;
+
+  /**
+   * PrimoStatementRenderer constructor.
+   *
+   * @param array $mapping
+   *   Mapping between common fields and primo fields.
+   */
+  public function __construct(array $mapping) {
+    $this->mapping = $mapping;
+  }
+
+  /**
+   * Renders a list of statements into primo briefsearch queries.
+   *
+   * @param array $statements
+   *   List of statements.
+   *
+   * @return array
+   *   The rendered statements as a list of <key> => query where key is a valid
+   *   Primo briefsearch query parameter.
+   *
+   * @throws \Ting\Search\UnsupportedSearchQueryException
+   *   If Primo cannot support the query.
+   */
+  public function renderStatements(array $statements) {
+    // Ensure we can support the statements and normalize the statements into
+    // a list of groups.
+    $groups = $this->preprocessStatements($statements);
+
+    // We now have a list of groups containing one or more fields each. If the
+    // group contains several statements, they will be against the same field.
+    // We can now map the array using $this->renderGroup();
+    return array_reduce(array_map([$this, 'renderGroup'], $groups), function ($carry, $rendered_group) {
+      foreach ($rendered_group as $parameter_key => $parameter_value) {
+        if(isset($carry[$parameter_key])) {
+          $carry[$parameter_key] = array_merge($carry[$parameter_key], $parameter_value);
+        }
+        else {
+          $carry[$parameter_key] = $parameter_value;
+        }
+      }
+      return $carry;
+    }, []);
+  }
+
+  /**
+   * Verify we can process the statements and convert them to groups.
+   *
+   * @param mixed[] $statements
+   *   One or more instances of BooleanStatementGroup and/or
+   *   TingSearchFieldFilter.
```

Comment:
This rubs be a bit the wrong way. I wonder if we could introduce a shared interface between the two.

Reply:
The inteface would look something like this (unless you're thinking a tag-interface?)
```
interface BooleanStatementMember {
   /**
   * ..
   * @return TingSearchFieldFilter[]
   */
}
```
What would that interface help me achieve? I still have to do an instanceof when handling the actual members of the array (as I want to treat groups differently from fields) so the interface is not helping me signal what you are allowed to pass in to the function as we only support known implementations of the interface.

I get that an "untyped" array is messy - but in this case I'm pretty clear about what I will allow in.

Reply:
Gave it another look, I'll implement a tag interface, but the bulk of the work has to happen in bbs-sal so I'm putting this PR on hold until I get that fixed and have bbs-master updated.